src/densities.jl

```diff
@@ -5,6 +5,17 @@
 #        compute_density, compute_δρ, compute_kinetic_energy_density
 #      - Use symmetrization instead of explicit use of symmetry operators
 
+function _check_positive(ρ)
+    minimum(ρ) < 0 && @warn("Negative ρ detected",
+                            min_ρ=minimum(ρ))
+end
+function _check_total_charge(basis, ρ, N)
+    n_electrons = sum(ρ) * basis.dvol
+    if abs(n_electrons - N) > sqrt(eps(eltype(basis)))
```

Comment:
T could be inferred from ρ and instead of passing basis, pass dvol (makes function more flexible)